web/webhook_test.go

```diff
@@ -261,6 +261,55 @@ func TestIncomingWebhook(t *testing.T) {
 		require.Nil(t, err)
 		assert.True(t, resp.StatusCode == http.StatusNotImplemented)
 	})
+
+	t.Run("IncomingWebhooksTriggerOutgoingWebhooks", func(t *testing.T) {
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.ServiceSettings.EnableIncomingWebhooks = true
+			*cfg.ServiceSettings.IncomingWebhooksTriggerOutgoingWebhooks = true
+			*cfg.ServiceSettings.AllowedUntrustedInternalConnections = "127.0.0.1"
+		})
+
+		called := make(chan bool)
+
+		go func(called chan bool) {
+			called <- false
+			http.HandleFunc("/", func(writer http.ResponseWriter, request *http.Request) {
+				called <- true
+				close(called)
+			})
+			err := http.ListenAndServe(":9123", nil)
+			require.Nil(t, err)
+		}(called)
+
+		_, appErr := th.App.CreateOutgoingWebhook(&model.OutgoingWebhook{
+			ChannelId:    th.BasicChannel.Id,
+			CreatorId:    th.BasicUser.Id,
+			TeamId:       th.BasicTeam.Id,
+			CallbackURLs: []string{"http://127.0.0.1:9123/"},
+		})
+		require.Nil(t, appErr)
+
+		text := `this is a \"test\"
+	that contains a newline and a tab`
+		resp, err := http.Post(url, "application/json", strings.NewReader("{\"text\":\""+text+"\"}"))
+		require.Nil(t, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+		timeout := time.After(2 * time.Second)
+		wait := true
+
+		for wait {
+			select {
+			case atLeastCalled := <-called:
+				if atLeastCalled {
+					wait = false
+				}
```

Comment:
I think we can just close the called channel, which will fire the select -- so no need to send false or true, and you don't need to check the result of the called channel (if it closes, it worked). 
(If I'm remembering correctly)

Reply:
Yes, you're right. Fixed that 👍

Reply:
Does the test currently work without the `called <- false` and `called <- true` lines? If it does, I think they should be removed and just have the `close()` call.

Reply:
I simplified the test so there is only the called <- true in the HandleFunc. Imho it is easier to check if the channel received a value than checking/waiting for it to be closed in this case.